plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -532,8 +532,10 @@ class Pub(
                     var vcs = VcsInfo.EMPTY
                     var authors = emptySet<String>()
 
+                    val source = pkgInfoFromLockFile["source"].textValueOrEmpty()
```

Comment:
How about doing the same for `pkgInfoFromLockFile["description"].textValueOrEmpty()`, too, while at it?

Reply:
On a second thought, maybe just extract `val description = pkgInfoFromLockFile["description"]` so you can also make use of it in your last commit.

Reply:
The reason is probably because there is already a `var description`, see line 529.

Reply:
Right... but that could easily be solved by using a different variable name, like `descriptionNode`. Anyway, it was just a proposal.

Reply:
I think the whole function needs a refactoring, especially the large when condition. But I don't have time to look deeper into it, esp. since the test data is incomplete, e.g. there are no tests for the `host == "git"` case, so I only did the small change that was useful for my change.